src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -273,6 +380,16 @@ func Register(
 		return *resErr
 	}
 
+	// Make sure normal user isn't registering under an exclusive application
+	// service namespace
+	if r.Auth.Type != "m.login.application_service" &&
+		cfg.Derived.ExclusiveApplicationServicesUsernameRegexp.MatchString(r.Username) {
+		return util.JSONResponse{
+			Code: 400,
+			JSON: jsonerror.Exclusive("This username is registered by an application service."),
```

Comment:
The username might not be registered, so we should probably have some different wording

Reply:
`jsonerror.Exclusive` is defined as:

> +// Exclusive is an error returned when an application service tries to register
+// an username that is outside of its registered namespace

which doesn't really fit

Reply:
Ah yes, the spec return `M_EXCLUSIVE` [for multiple reasons](https://matrix.org/docs/spec/application_service/unstable.html#server-admin-style-permissions). I'll update the comments to reflect that :)